src/components/search-page/stores/__tests__/SearchLocalStore.test.ts

```diff
@@ -0,0 +1,40 @@
+import { SearchLocalStore, SearchLocalStoreType } from '../SearchLocalStore';
+import environment from 'environment';
+import { initialResponseStatus } from '../../../../models/IResponseStatus';
+import nock from 'nock';
+
+describe('SearchLocalStore', () => {
+  let searchLocalStore: SearchLocalStoreType;
+
+  beforeEach(() => {
+    searchLocalStore = SearchLocalStore({ endpoint: environment.api.showsSearch });
+  });
+
+  test('should have default state', () => {
+    expect(searchLocalStore.endpoint).toEqual(environment.api.showsSearch);
+    expect(searchLocalStore.currentSearchTerm).toEqual('');
+    expect(searchLocalStore.searchResults).toEqual(initialResponseStatus([]));
+    expect(searchLocalStore.resultsText).toEqual('Results: 0');
+  });
+
+  test('should start search term request', () => {
+    // 'https://api.tvmaze.com/search/shows?q=Friends'
+
+    nock('https://api.tvmaze.com')
+      .get('/search/shows?q=Friends')
+      .reply(200, {
+        results: 'any thing so that I can see this fail at least',
+      });
```

Comment:
@n8rzz I need help I can't get `nock`

Reply:
Maybe none of the axios request are being made within jest? If I remove nock and check if the request was actually made I don't get anything even if I add a try/catch within `requestAction`. No code gets called after the await. https://github.com/codeBelt/nextjs-mobx/blob/master/src/utilities/mobxUtil.ts#L19

Reply:
how does `async` work with mobx here?  `.serach()` isn't listed as async, however it does end up calling `._requestData()` which is.  perhaps that's causing issues.

everything else checks out.  your tests are passing (for me at least) and everything seems to be working ok, minus the async item above.  One other thing to call out, `nock` only _nocks_ one time.  So you may want to abstract anything nock related into a function that runs the nock stuff and can be easily repeated.

Reply:
I don't think `nock` is working. I update this test with how I would like it to fully work. Notice the before and after comments. `nock` is not returning the `expectedData`. What I am getting is an HttpErrorResponseModel and don't know why when `nock` should be catching the request

Reply:
This fixed my issue: https://github.com/axios/axios/issues/1754#issuecomment-435784235